### ISSUE TITLE
LexRuntimeServiceClient::PostContent seems broken in current code

### ISSUE THREAD

**Issue by BitStarUK** [aws/aws-sdk-cpp#1152]
LexRuntimeServiceClient::PostContent seems broken in current code.

LexRuntimeServiceClient::PostContent() always returns UNSUPPORTED_MEDIA_TYPE error as all HTTP request headers seem to be sent with Content Type 'application/vnd.amazon.eventstream' instead of whatever is set with request.SetContentType().

Tracing through the code with the debugger the issue seems to be in inline AmazonStreamingWebServiceRequest::GetHeaders() which attempts to 'insert' the correct content type...

`headers.insert(Aws::Http::HeaderValuePair(Aws::Http::CONTENT_TYPE_HEADER, GetContentType()));`

... but the std::map 'headers' already contains an entry for this so the insert does nothing, resulting in all http requests being sent with the default 'application/vnd.amazon.eventstream' which is set with emplace in PostContentRequest::GetRequestSpecificHeaders().

You can use the following code to reproduce...

```
        Aws::LexRuntimeService::Model::PostContentRequest request;

	request.SetBotName("TestBot");		// Dummy name for test

	request.SetBotAlias("$LATEST");
	request.SetUserId("TestUser");
	request.SetAccept("audio/pcm");

	request.SetContentType("audio/l16; rate=16000; channels=1");

	// For test, no need to set body with audio as it's ignored due to content type error

	Aws::LexRuntimeService::Model::PostContentOutcome outcome = lexClient->PostContent(request);

	if (outcome.IsSuccess()) {
		std::cout << "PostContent: Success"
			<< std::endl;
	}
	else {
		auto error = outcome.GetError();
		std::cout << "ERROR: " << error.GetExceptionName() << ": "
			<< error.GetMessage() << std::endl;
	}
```

  
... then put a breakpoint in AmazonStreamingWebServiceRequest::GetHeaders() and see that the correct content type never gets set.

I can fix it locally by changing AmazonStreamingWebServiceRequest::GetHeaders to...

```
       inline Aws::Http::HeaderValueCollection GetHeaders() const override
        {
            auto headers = GetRequestSpecificHeaders();
	    headers[Aws::Http::CONTENT_TYPE_HEADER] = GetContentType();

            return headers;
        }  
```
then everything works fine.

Is this broken or am I doing something wrong?

**Comment by wps132230** [comment]
Thank you for reporting this issue.
We can confirm it's a bug, and will provide a fix soon.

**Comment by marcomagdy** [comment]
Fixed in v1.7.116
